RadixWallet/Features/SecurityCenterFeature/SecurityFactors/FactorSourcesList.swift

```diff
@@ -196,23 +228,73 @@ private extension FactorSourcesList {
 				// source was backed up.
 				.notBackedUp
 			}
+
+			// Determine row selectability
+			let selectability: State.Row.Selectability = if status == .lostFactorSource {
+				.invalid
+			} else if selectedFactorSourceIds.contains(entity.integrity.factorSource.id) {
+				.alreadySelected
+			} else {
+				.selectable
+			}
 			return State.Row(
 				integrity: entity.integrity,
 				linkedEntities: entity.linkedEntities,
-				status: status
+				status: status,
+				selectability: selectability
 			)
 		}
 	}
+
+	func getSelectedFactorSourceIds(state: State, factorSourceIds: [FactorSourceId]) -> [FactorSourceId] {
```

Comment:
this method has a confusing name IMO. It actually filters. So its name ought to contain "filter" or "filtered". Maybe:

`filterValidFactorSourceIds`?

I dont thinkg `selected` should be in the name. The function is pure, right?